### PR TITLE
RHCLOUD-22943 Migrate from endpoints ordinal to name in the DB (step 1)

### PR DESCRIPTION
**Pull request by gwenneg**
This PR contains step 1 of the migration from endpoints `ordinal` to `name` in the DB. It prepares the DB for a subsequent migration of the Java code which won't require any maintenance mode.

### CODE REVIEW

File: database/src/main/resources/db/migration/V1.70.0__RHCLOUD-22943_no_more_endpoint_type_ordinal.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
+-- This is step 1 of the migration from endpoints types "ordinal" to "name".
+
+ALTER TABLE endpoints
+    ADD COLUMN endpoint_type_v2 VARCHAR(20);
```
Comment (gwenneg) [inline database/src/main/resources/db/migration/V1.70.0__RHCLOUD-22943_no_more_endpoint_type_ordinal.sql:4]: Feel free to suggest a better name than `endpoint_type_v2` 😄

File: database/src/main/resources/db/migration/V1.70.0__RHCLOUD-22943_no_more_endpoint_type_ordinal.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+ALTER TABLE notification_history
+    ADD COLUMN endpoint_type_v2 VARCHAR(20);
+
+-- TODO Remove this function after the migration is done.
+CREATE FUNCTION transition_to_string_endpoint_type() RETURNS TRIGGER AS $$
+BEGIN
+    -- The endpoint type will be copied into the new column every time a row is inserted or updated.
+    NEW.endpoint_type_v2 := NEW.endpoint_type;
```
[the remark this answers is not in the document]
Comment (gwenneg) [reply database/src/main/resources/db/migration/V1.70.0__RHCLOUD-22943_no_more_endpoint_type_ordinal.sql:13]: Good catch! I'm battling with a stupid Hibernate bug in step 2... I should have self-reviewed this one first 😄

[the remark this answers is not in the document]
Comment (gwenneg) [reply database/src/main/resources/db/migration/V1.70.0__RHCLOUD-22943_no_more_endpoint_type_ordinal.sql]: Fixed!

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**